cli/azd/pkg/environment/environment.go

```diff
@@ -36,14 +37,78 @@ const ContainerRegistryEndpointEnvVarName = "AZURE_CONTAINER_REGISTRY_ENDPOINT"
 const ResourceGroupEnvVarName = "AZURE_RESOURCE_GROUP"
 
 type Environment struct {
-	// Values is a map of setting names to values.
-	Values map[string]string
+	// values is a map of setting names to values.
+	// all keys are converted to upper case before it is set.
+	values map[string]string
 	// File is a path to the file that backs this environment. If empty, the Environment
 	// will not be persisted when `Save` is called. This allows the zero value to be used
 	// for testing.
 	File string
 }
 
+// DeleteVariable removes an entry from azd environment which key matches the variable name.
+func (e *Environment) DeleteVariable(variableName string) {
```

Comment:
Have you considered dropping "Variable" and going with `Delete`, `Set` and `Get`? I think it's clear we're setting a key-value pair / variable. A good point to support this is we don't add "Variable" prefix consistently, i.e. we don't say: `ValueOfVariable` or `GetValueOfVariable`

Reply:
yes, but I don't feel 100% happy when I read `env.Delete()`, as it makes me think that it will Delete the environment :(

I think I can move all the CRUD operations as part of an extra structure inside the environment that wraps the map[string]string.  So we can do something like `env.Values.Delete(...)`, which would make me happier

Reply:
What about `Unset`?